packages/server-core/src/user/identity-provider/identity-provider.test.ts

```diff
@@ -69,37 +70,61 @@ describe('identity-provider service', () => {
       {
         type,
         token,
-        password
+        password,
+        userId
       },
       {}
     )
-    providers.push(item)
 
     assert.equal(item.type, type)
     assert.equal(item.token, token)
     assert.ok(item.userId)
   })
 
   it('should find identity providers', async () => {
-    for (const provider of providers) {
-      const item = await app.service('identity-provider').find({
-        query: {
-          userId: provider.userId
-        }
-      })
-
-      assert.ok(item, 'Identity provider item is found')
-    }
+    const item = await app.service('identity-provider').find({
+      query: {
+        userId
+      }
+    })
+
+    assert.ok(item, 'Identity provider item is found')
   })
 
   it('should remove identity providers', async () => {
```

Comment:
Looking at this test, I would like it to be two tests. One should test removing one identity-provider by ID, and should check that that i-p is no longer present while the other are. The second would remove by userId as this test is, and does an explicit check that there are no more identity-providers for that user.

Reply:
Added this change!